docs/geometry.rst

```diff
@@ -0,0 +1,132 @@
+********
+Geometry
+********
+
+Manifolds
+=========
+
+Manifold
+--------
+.. automodule:: geomstats.geometry.manifold
+   :members:
+
+Embedded Manifold
+-----------------
+.. automodule:: geomstats.geometry.embedded_manifold
+  :members:
+
+Product Manifold
+----------------
+.. automodule:: geomstats.geometry.product_manifold
+   :members:
+
+Euclidean Space
+---------------
+.. automodule:: geomstats.geometry.euclidean
+   :members:
+
+Minkowski Space
+---------------
+.. automodule:: geomstats.geometry.minkowski
+   :members:
+
+Hypersphere
+-----------
+.. automodule:: geomstats.geometry.hypersphere
+   :members:
+
+Hyperbolic Space
+----------------
+.. automodule:: geomstats.geometry.hyperbolic
+   :members:
+
+Discretized Curves Space
+------------------------
+.. automodule:: geomstats.geometry.discretized_curves
+   :members:
+
+Symmetric-Positive-Definite (SPD) Matrices Space
+------------------------------------------------
+.. automodule:: geomstats.geometry.spd_matrices
+   :members:
+
+Skew-Symmetric Matrices Space
+-----------------------------
+.. automodule:: geomstats.geometry.skew_symmetric_matrices
+   :members:
+
+Stiefel Manifold
+----------------
+.. automodule:: geomstats.geometry.stiefel
+   :members:
+
+Lie Groups
+----------
+.. automodule:: geomstats.geometry.lie_group
+   :members:
+
+Lie Algebra
```

Comment:
Does it make sense to put Lie Algebra (and Matrices space, and Skew Symmetric Matrices) in Manifolds?

Reply:
Good question. Since the doc is generated automatically, did it put it there because of our inheritance structure?

If we want to change the API doc, we probably need to change the code first.

Reply:
I did that. I wasn't sure what the best categorization scheme was for the geometry section. Given that they all inherit Manifold, I put them in that section for now; please update as you see fit.

I wish those were generated automatically; perhaps in the future they will be.